File: portlets/sample-python-portlet/docroot/WEB-INF/python/common/Custom.py
from javax.portlet.PortletPreferences import *;
from javax.portlet.RenderRequest import *;
from javax.portlet.RenderResponse import *;

class Custom:
    def __init__(self, renderRequest, renderResponse):
        self.renderRequest = renderRequest;
        self.renderResponse = renderResponse;
        self.lp = LiferayPortlet(renderRequest, renderResponse);

    def sayHello(self):
        return """
            <div>
                <b>Hello Python Portlet users!</b>
            </div>
        """;
    
    def navigation(self):
        return '\n' +
            '<div class="separator"></div>\n\n' +
            '<div>\n' +
                'Navigation:\n' +
                '\n' + self.lp.renderLink("Home", "/WEB-INF/python/view.py", {}) + " - " + 
                self.lp.renderLink("User Info", "/WEB-INF/python/info.py", {'type' : 'user'}) + " - " + 
                self.lp.renderLink("Portlet Info", "/WEB-INF/python/info.py", {'type' : 'portlet'}) + " - " + 
                self.lp.renderLink("Invoke Java", "/WEB-INF/python/java.py", {}) + '\n' +
            '</div>\n';
    
    def showNumber(self):
        preferences = self.renderRequest.getPreferences();
    
        num = preferences.getValue("number", "0");
        actionLink = self.lp.actionLink("Increment", {});
    
        return '<div>#Number: ' + num + ' (' + actionLink + ')</div>';

File: portlets/sample-ruby-portlet/docroot/WEB-INF/ruby/global/custom.rb
module Custom
	def Custom.navigation
		"""
			<div class='separator'></div>

			<div>
				Navigation:

				#{LiferayPortlet.renderLink("Home", "/WEB-INF/ruby/view.rb")} -
				#{LiferayPortlet.renderLink("User Info", "/WEB-INF/ruby/info.rb", {'type' => 'user'})} -
				#{LiferayPortlet.renderLink("Portlet Info", "/WEB-INF/ruby/info.rb", {'type' => 'portlet'})} -
				#{LiferayPortlet.renderLink("Invoke Java", "/WEB-INF/ruby/java.rb")}
			</div>
		"""
	end

	def Custom.sayHello
		"""
			<div>
				<b>Hello Ruby Portlet users!</b>
			</div>
		"""
	end

	def Custom.showNumber(renderRequest)
		preferences = renderRequest.getPreferences();
		num = preferences.getValue("number", "0")

		"""
			<div>
				Number: #{num} (#{LiferayPortlet.actionLink("Increment")})
			</div>
		"""
	end
end